lib/sqlalchemy/sql/compiler.py

```diff
@@ -5166,8 +5179,29 @@ def get_column_specification(self, column, **kwargs):
             not column.identity or not self.dialect.supports_identity_columns
         ):
             colspec += " NOT NULL"
+
+        if column.system_versioning:
+            option = self.get_column_versioning_options(self, column)
+            colspec += " %s" % option
+
         return colspec
 
+    def get_column_versioning_options(self, *args, **kw) -> str:
+        """Return options for system versioning
+        Allows enabling or disabling, or configuring "GENERATED" statements"""
+        column = args[1]
+        if column.system_versioning == False:
+            return "WITHOUT SYSTEM VERSIONING"
+        if column.system_versioning == True:
+            return "WITH SYSTEM VERSIONING"
+        elif column.system_versioning == "start":
+            column.table._versioning_columns["start"] = column.name
```

Comment:
we can't make any mutations to a Table  / Column during compilation, so all of this state change would need to happen when Period is assigned to a Table.

Reply:
Sorry, I think this was a holdover from an old implementation I played around with but didn't remove yet

Reply:
This is gone